### PR TITLE
[C API] Add support for setting/getting new nneg flag on zext instructions

### PR DESCRIPTION
**Pull request by Benjins**
This flag was added in #67982, but was not yet accessible via the C API. This commit adds a getter/setter for this flag, and a test for it

### CODE REVIEW

**Review by nikic — APPROVED** [review]
LGTM

**Comment by Benjins** [comment]
Not sure what the issue with the build is: the only test failure is in the Window's build in `TEST 'MLIR :: python/dialects/scf.py'`, which shouldn't be affected by these changes. I can try merging from main if we want to make it green before landing, otherwise is this good to merge in?

**Comment by Benjins** [comment]
Bumping this: if it's good to land, can someone with write access merge it?

**Pull request merged**
